pay-api/tests/unit/services/test_direct_pay_service.py

```diff
@@ -257,5 +260,172 @@ def test_process_cfs_refund_duplicate_refund(monkeypatch):
+                'linenumber': '1',
+                'revenueaccount': '112.32363.34725.4375.3200062.000000.0000',
+                'revenueamount': '130',
+                'glstatus': 'CMPLT',
+                'glerrormessage': None,
+                'refund_data': []
+            },
+            {
+                'linenumber': '2',
+                'revenueaccount': '112.32363.34725.4375.3200054.000000.0000',
+                'revenueamount': '1.5',
+                'glstatus': 'CMPLT',
+                'glerrormessage': None,
+                'refund_data': [],
+                'not_part_of_spec': 'heyey'
+            }
+        ],
+        'postedrefundamount': None,
+        'refundedamount': None
+    }
+    paybc_order_status = Converter().structure(paybc_response, OrderStatus)
+    assert paybc_order_status
+
+
+def _automated_refund_preparation():
+    """Help automated_refund tests below."""
+    payment_account = factory_payment_account()
+    invoice = factory_invoice(payment_account, total=30, service_fees=1.5)
+    invoice.invoice_status_code = InvoiceStatus.PAID.value
+    invoice.payment_date = datetime.now()
+    invoice.save()
+    payment_line_item = factory_payment_line_item(invoice.id, fee_schedule_id=1, service_fees=1.5, total=30)
+    payment_line_item.save()
+    receipt = factory_receipt(invoice.id, invoice.id, receipt_amount=invoice.total).save()
+    receipt.save()
+    invoice_reference = factory_invoice_reference(invoice.id, invoice.id)
+    invoice_reference.status_code = InvoiceReferenceStatus.COMPLETED.value
+    invoice_reference.save()
+    return invoice, payment_line_item
+
+
+@pytest.mark.parametrize('test_name, refund_partial', [
+    ('pay_pli_not_exist', [
+        RefundPartialLine(payment_line_item_id=1,
+                          refund_amount=1,
+                          refund_type=RefundsPartialType.OTHER_FEES.value)
+    ]),
+    ('pay_negative_refund_amount', [
+        RefundPartialLine(payment_line_item_id=1,
+                          refund_amount=-1,
+                          refund_type=RefundsPartialType.OTHER_FEES.value)]),
+    ('pay_service_fee_too_high', [
+        RefundPartialLine(payment_line_item_id=1,
+                          refund_amount=3,
+                          refund_type=RefundsPartialType.SERVICE_FEE.value)]),
+])
+def test_build_automated_refund_payload_validation(test_name, refund_partial):
+    """Assert validations are working correctly for building refund payload."""
+    invoice, payment_line_item = _automated_refund_preparation()
+    if test_name == 'pay_pli_not_exist':
+        refund_partial[0].payment_line_item_id = 999999999
+    else:
+        refund_partial[0].payment_line_item_id = payment_line_item.id
+    direct_pay_service = DirectPayService()
+    with pytest.raises(BusinessException) as excinfo:
+        direct_pay_service.build_automated_refund_payload(invoice, refund_partial)
+        assert excinfo.value.code == Error.INVALID_REQUEST.name
```

Comment:
I wanted to return real errors, not sure your thoughts about returning INVALID REQUEST, on the plus side at least I wrote some logging to handle the cases

Reply:
This works for me